src/ddmd/dscope.d

```diff
@@ -556,7 +557,7 @@ struct Scope
                 if (!s)
                     s = searchScopes(flags | SearchImportsOnly | IgnoreSymbolVisibility);
 
-                if (s && !(flags & IgnoreErrors))
+                if (s && !(flags & IgnoreErrors) && !cachedSemantics)
                     .deprecation(loc, "%s is not visible from module %s", s.toPrettyChars(), _module.toChars());
```

Comment:
That looks fishy, you want to create a new nested scope and set SCOPEignoresymbolvisibility instead of using `cachedSemantics`.
https://github.com/dlang/dmd/blob/51bc2fb427a5e0b3701b11ab2b6a5f356d8eca5d/src/ddmd/traits.d#L733-L736